quickwit/quickwit-ingest/src/ingest_v2/router.rs

```diff
@@ -169,7 +170,9 @@ impl IngestRouter {
 
         let mut state_guard = self.state.lock().await;
 
-        for subrequest in pending_subrequests(&workbench.subworkbenches) {
+        for subrequest in
+            pending_subrequests_for_attempt(&workbench.subworkbenches, workbench.num_attempts)
```

Comment:
we added the logic to not process further subrequests for which we already oberved an error when trying to create the shards. Otherwise control plane errors are overriden as "no shard available".